frontend/src/utils/main.js

```diff
@@ -8,3 +9,16 @@ export const getCollabRoomId = (roomId) =>
 // returns a date object representing the timestamp 15 mins from now
 export const getJWTExpiry = () =>
   new Date(new Date().getTime() + JWT_EXPIRY_IN_MINS * 60 * 1000)
+
+
+export function expirationCheck(datetime, expirationFunction) {
+  const now = moment()
+  const expiration = moment(datetime)
+
+  // get the difference between the moments of the two dates in minutes
+  const diff = now.diff(expiration, 'minutes')
+  console.log("diff  bro: ", diff)
+
+  expirationFunction(diff)
+  
+}
```

Comment:
```suggestion
export function expirationCheck(datetime, expiredCallback, notExpiredCallback) {
  const now = moment()
  const expiration = moment(datetime)

  // get the difference between the moments of the two dates in minutes
  const diff = now.diff(expiration, 'minutes')
  if (diff < 30) {
    notExpiredCallback()
  } else {
    expiredCallback()
  }
}
```

I meant for the 2 callbacks to be passed in here 😅

Otherwise the calling code still duplicates the expiration check logic. For instance, if we change the expiry to 45mins next time, notice that we'll have to hunt down and change all the places where we do the check since it's not abstracted here

Reply:
that's true! Will make the changes before merging!